MantidQt/CustomInterfaces/src/Indirect/ISISDiagnostics.cpp

```diff
@@ -467,5 +467,23 @@ void ISISDiagnostics::pbRunFinished() {
   m_uiForm.dsInputFiles->setEnabled(true);
 }
 
+/**
+ * Handles mantid plotting
+ */
+void ISISDiagnostics::plotClicked() {
+  bool plot = checkADSForPlotSaveWorkspace(m_pythonExportWsName, true);
+  if (plot)
+    plotSpectrum(QString::fromStdString(m_pythonExportWsName));
+}
+
+/**
+ * Handles saving workspace
+ */
+void ISISDiagnostics::saveClicked() {
+  bool save = checkADSForPlotSaveWorkspace(m_pythonExportWsName, false);
+  if (save)
```

Comment:
This can be turned into
`if (checkADSForPlotSaveWorkspace(m_pythonExportWsName, false)` so we dont need `save`